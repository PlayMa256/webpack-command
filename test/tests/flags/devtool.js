const { existsSync: exists } = require('fs');
const { resolve } = require('path');

const { apply, build, crcDist, test, validate } = require('../../util');

const fixture = 'devtool/devtool';
const opts = { fixture };

let config;

test('--devtool', module, () => {
  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should build`, () =>
    build(config).then((result) => {
      const mapPath = resolve(__dirname, '../../../dist/main.js.map');
      expect(exists(mapPath));
      expect(result).toMatchSnapshot();
      expect(crcDist()).toMatchSnapshot();
      expect(crcDist(mapPath)).toMatchSnapshot();
    }));
});
