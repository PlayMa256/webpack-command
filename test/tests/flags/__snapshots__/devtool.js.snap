// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --devtool > should apply #0 1`] = `
Object {
  "devtool": "cheap-source-map",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;

exports[`Flags > --devtool > should build #0 1`] = `
"Version: webpack 4.12.1
      Asset      Size  Chunks             Chunk Names
    main.js  3.88 KiB    main  [emitted]  main
main.js.map  3.59 KiB    main  [emitted]  main
Entrypoint main = main.js main.js.map
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --devtool > should build #1 1`] = `"dc2cf76c"`;

exports[`Flags > --devtool > should build #2 1`] = `"dccd936"`;
